lib/ws/ws.js

```diff
@@ -466,7 +466,7 @@ BeamSocket.prototype.call = function (method, args_, options_) {
             self._replies[id] = new Reply(resolve, reject);
         }),
     ])
-    .catch(TimeoutError, function (err) {
+    .catch(function (err) {
         delete self._replies[id];
```

Comment:
Probably

```js
if (err instanceof TimeoutError) {
  delete self._replies[id];
}
```

If a rejection occurs on the _reply_ it will already have been deleted from the registry, you just need to catch timeout errors.